src/radical/pilot/agent/scheduler/base.py

```diff
@@ -653,6 +640,10 @@ def schedule_cb(self, topic, msg):
                 with self._wait_lock:
                     self._wait_pool.remove(unit)
 
+            else:
```

Comment:
Why it should break if the workload is uniform?

Reply:
see above :-)  A new iteration cannot yield a different result.